lib/dxf/entity.rb

```diff
@@ -147,10 +147,25 @@ def initialize(*points)
 
     # Return the individual line segments
     def lines
-      points.each_cons(2).map { |a, b| Line.new a, b }
+      points.each_cons(2).map do |a, b|
+        line = Line.new a,b
+        line.layer = self.layer
+        line.color_number = self.color_number
+        line.x1 = a[0]
+        line.y1 = a[1]
+        line.x2 = b[0]
+        line.y2 = b[1]
+        line
+      end
```

Comment:
This change is added to include the values for layer, color and x, y coordinates on the Line object being returned.